src/ynEIGEN/WithdrawalsProcessor.sol

```diff
@@ -0,0 +1,432 @@
+// SPDX-License-Identifier: BSD 3-Clause License
+pragma solidity ^0.8.24;
+
+import {IERC20} from "lib/openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+import {IERC4626} from "lib/openzeppelin-contracts/contracts/interfaces/IERC4626.sol";
+import {AccessControlUpgradeable} from
+    "lib/openzeppelin-contracts-upgradeable/contracts/access/AccessControlUpgradeable.sol";
+import {Initializable} from "lib/openzeppelin-contracts-upgradeable/contracts/proxy/utils/Initializable.sol";
+
+import {IDelegationManager} from "lib/eigenlayer-contracts/src/contracts/interfaces/IDelegationManager.sol";
+import {IStrategy} from "lib/eigenlayer-contracts/src/contracts/interfaces/IStrategy.sol";
+
+import {IwstETH} from "../external/lido/IwstETH.sol";
+
+import {IWithdrawalQueueManager} from "../interfaces/IWithdrawalQueueManager.sol";
+import {ITokenStakingNodesManager} from "../interfaces/ITokenStakingNodesManager.sol";
+import {ITokenStakingNode} from "../interfaces/ITokenStakingNode.sol";
+import {IAssetRegistry} from "../interfaces/IAssetRegistry.sol";
+import {IYieldNestStrategyManager} from "../interfaces/IYieldNestStrategyManager.sol";
+import {IynEigen} from "../interfaces/IynEigen.sol";
+import {IRedemptionAssetsVault} from "../interfaces/IRedemptionAssetsVault.sol";
+import {IWrapper} from "../interfaces/IWrapper.sol";
+import {IWithdrawalsProcessor} from "../interfaces/IWithdrawalsProcessor.sol";
+
+contract WithdrawalsProcessor is IWithdrawalsProcessor, Initializable, AccessControlUpgradeable {
+
+    uint256 public totalQueuedWithdrawals;
+
+    uint256 public minPendingWithdrawalRequestAmount;
+
+    IDs private _ids;
+
+    mapping(uint256 id => QueuedWithdrawal) private _queuedWithdrawals;
+    mapping(uint256 fromId => uint256 toId) public batch;
+
+    // yieldnest
+    IWithdrawalQueueManager public immutable withdrawalQueueManager;
+    ITokenStakingNodesManager public immutable tokenStakingNodesManager;
+    IAssetRegistry public immutable assetRegistry;
+    IYieldNestStrategyManager public immutable ynStrategyManager;
+    IynEigen public immutable yneigen;
+    IRedemptionAssetsVault public immutable redemptionAssetsVault;
+    IWrapper public immutable wrapper;
+
+    // eigenlayer
+    IDelegationManager public immutable delegationManager;
+
+    // assets
+    IERC20 private constant STETH = IERC20(0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84);
```

Comment:
the hardcoded mainnet addresses won't work for holesky

Reply:
on holesky we really just have steth from the 2, there's no testnet OETH

Reply:
oh right good point

Reply:
https://github.com/yieldnest/yieldnest-protocol/pull/190/commits/ec1c596d931ef7b513e6b26b511cbf19128af524